components/service/glean/src/test/java/mozilla/components/service/glean/scheduler/MetricsPingSchedulerTest.kt

```diff
@@ -54,18 +58,13 @@ class MetricsPingSchedulerTest {
         return null as T
     }
 
-    @Before
-    fun setup() {
-        WorkManagerTestInitHelper.initializeTestWorkManager(
-            ApplicationProvider.getApplicationContext())
-
-        Glean.enableTestingMode()
-    }
+    @get:Rule
+    val gleanRule = GleanTestRule(ApplicationProvider.getApplicationContext())
```

Comment:
I think this might end up causing a few more intermittents: we init the scheduler in this rule as well. This is the reason why this test wasn't using the `GleanTestRule` in the first place. I'd recommend adding a test-only `testReset` method to `MetricsPingScheduler` that clears whatever SharedPrefs it uses and then call it at the beginning of the relevant tests.

Reply:
Just one and it's not intermittent, it's always on.  I'm refactoring this now to create a method specific reset as you suggest.